File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/event/PlayerWorldChangeEvent.java
package top.likoslupus.cellulosesz.api.event;

import top.likoslupus.cellulosesz.api.platform.CellPlayer;

import static java.util.Objects.requireNonNull;

public record PlayerWorldChangeEvent(
        CellPlayer player,
        String fromWorld,
        String toWorld
) {

    public PlayerWorldChangeEvent {
        requireNonNull(player, "player");
        requireNonNull(fromWorld, "fromWorld");
        requireNonNull(toWorld, "toWorld");
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/event/PlayerWorldChangeEvent.kt
package top.likoslupus.cellulosesz.api.event

import top.likoslupus.cellulosesz.api.platform.CellPlayer

@JvmRecord
public data class PlayerWorldChangeEvent(
    public val player: CellPlayer,
    public val fromWorld: String,
    public val toWorld: String
)
